src/contexts/filterContext.js

```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+import { createContext, useContext, useReducer, useEffect } from "react";
+import { filterReducer } from '../reducers/filterReducer';
+
+const FilterContext = createContext();
+const useFilter = () => useContext(FilterContext);
+const initialState =  {
+    sortBy: "",
+    categories: {},
+    priceRange: 600,
+    starRating: "",
+    products:[]
+  };
+const FilterContextProvider = ({ children }) => {
+    const [state, dispatch] = useReducer(filterReducer, initialState);
+
+    useEffect(() => {
+        (async () => {
+            try{
+                const {data : {categories}} = await axios.get('/api/categories');
+                dispatch({
+                  type: "INIT_CATEGORIES", 
+                  payload: categories
+                });
+            }catch(error)
+            {
+              console.log(error);
+            }
+        })();
+    }, []);
+
+    useEffect(() => {
+      (async () => {
+        try{
+          const {data : {products}} = await axios.get('/api/products');
```

Comment:
you can call the products API on only the product page and product listing page as apart from that the data of a product will be stored in the context or local storage for further use. calling multiple API in the initiative would lead to slow web performance and also by this you have to take care of so much data.

Reply:
Thanks for the suggestion Vinayak. I'll find a way to call the product api from product page.